rkt/image_cat_manifest.go

```diff
@@ -49,31 +50,38 @@ func runImageCatManifest(args []string) (exit int) {
 
 	s, err := store.NewStore(globalFlags.Dir)
 	if err != nil {
-		stderr("image cat-manifest: cannot open store: %v\n", err)
+		stderr("image cat-manifest: cannot open store: %v", err)
 		return 1
 	}
-	ks := getKeystore()
 
-	fn := &finder{
-		imageActionData: imageActionData{
-			s:                  s,
-			ks:                 ks,
-			insecureSkipVerify: true,
-			debug:              globalFlags.Debug,
-		},
-		local:    true,
-		withDeps: false,
-	}
-
-	h, err := fn.findImage(args[0], "", true)
-	if err != nil {
-		stderr("image cat-manifest: cannot find image: %v\n", err)
-		return 1
+	var key string
+	if _, err := types.NewHash(args[0]); err == nil {
+		key, err = s.ResolveKey(args[0])
+		if err != nil {
+			stderr("image cat-manifest: cannot resolve key: %v", err)
+			return 1
+		}
+	} else {
+		app, err := discovery.NewAppFromString(args[0])
+		if err != nil {
+			stderr("image cat-manifest: cannot parse the image name: %v", err)
+			return 1
+		}
+		labels, err := types.LabelsFromMap(app.Labels)
+		if err != nil {
+			stderr("image cat-manifest: invalid labels in the name: %v", err)
+			return 1
+		}
+		key, err = s.GetACI(app.Name, labels)
+		if err != nil {
+			stderr("image cat-manifest: cannot find image: %v %v", err, args[0])
```

Comment:
+1; although for consistency with the other error messages, probably just drop the name from this one entirely